### ISSUE TITLE
Valhelsia 3 Version 3.1.2 Crashing when Chunk Loading and General Gameplay

### ISSUE THREAD

**Issue by TheWolf531** [jediminer543/JMT-MCMT#41]
On version 1.16.4, mostly found when loading new chunks, but several crashes were also in general gameplay. I removed performant and yet the crashing continued. Another issue, the server tickrate with performant installed dropped significantly with the world option of mcmt toggled on, after removing performant or disabling world, worked fine until crashes. Attached are several crash reports.

https://pastebin.com/zN4tAd2L

This first one crashed while loading new chunks, and have had several crash reports not unlike this one, looking into them I saw the chunkprovider was mentioned a lot, tried turning that off with MCMT, the crashes persisted.

https://pastebin.com/WM376n44

This one is quite similar to the first, and seems as if they are the same cause here. 

https://pastebin.com/fGHC2yd9

There is a third, I believe this one was after the removal of performant.

**Comment by NXTler** [comment]
Velhelsia 3 is a big modpack and what effects mcmt have on mods is still unclear. It could be difficult to find the issue since I know from my mod testing in the 1.15.2 (#35) that their can be issues with mod combinations but not with the stand alone mods and weird stuff like getting collision in spector mode with other blocks ect..

As I know @jediminer543 will fix the normal issues that mcmt have without any other mods. Than their will be a lot like really heavy mod testing to get it officiel compatible with most commen mods and this will take some time.

**Comment by TheWolf531** [comment]
Yes, I completely understand that, however I was just wondering if this might have been an issue with the mod directly rather than its interference with other mods, that's all.

**Comment by jediminer543** [comment]
This is kind of an issue with this mod rather than with the interference itself, but it only occurs in the presence of other mods

Basicly what happens is the chunk provider attempts to load a chunk and gets stuck for some reason. As the server is still waiting nothing happens. On dedicated clients there is a major issue with this as it this as it just stays locked up. On servers this results in the Watchdog killing the server.

`java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)`

This has been what I've been trying to work on the most, but it's an absolute PITA to debug as I kind of need access to the full state of the game

**Comment by TheWolf531** [comment]
So if the chunk provider is the part that gets stuck, would disabling the multithreaded chunk provider be a band-aid fix for the time being?

**Comment by TheWolf531** [comment]
Ah, shame. Thanks for the info though; have you thought about setting up a patreon? I'd really like to support you and the development of this amazing mod.

**Comment by BVHagop** [comment]
So I was trying to get this to work, and what I was seeing was about a 1000x increase in entity spawning. standing near an unlit cave would cause 400+ zombies to spawn in about 5 mins.

**Comment by BVHagop** [comment]
also, one beehive near my base produced about 100 bees at a time instead of one.

**Comment by TheWolf531** [comment]
Yeah I noticed a similar behavior, I initially used the mod during the start of the modpack so killing all the entities that weren't players was actually an option, although I'm not too sure how you would come to solve that issue without that. When I was still using this mod on the server, I found that with performant on alongside the MCMT mod, as long as the world multithreading was on the massive amount of mobs would spawn and the server's TPS would slowly decrease from 20.

**Comment by KubiRemPL** [comment]
Yeah, I just tested. I was doing constant "/forge entity list" and after a while of flying I saw an error and after that I done "/forge entity list" and BOOM..... I saw 1000's of bees and my 16 core processor was crying (to this point I needed to kill it). The error is pretty similar to this sended above (I will not show it here... but If you want to I will). And all was fixed by disabling world threading. I was using performant and MCMT on latest version (MC 1.16.5).

Best Regards,
Kubi
